pyteal/ast/abi/util_test.py

```diff
@@ -771,3 +771,79 @@ def test_sdk_type_specs_from_signature(sig_str, sig_args, sig_rets):
     args, ret = abi.type_specs_from_signature(sig_str)
     assert args == sig_args
     assert ret == sig_rets
+
+
+class NamedTDecl(abi.NamedTuple):
+    a: abi.Field[abi.Uint64]
+    b: abi.Field[
+        abi.Tuple3[abi.PaymentTransaction, abi.Address, abi.StaticBytes[Literal[16]]]
+    ]
+    c: abi.Field[abi.Transaction]
+
+
+TYPE_SPEC_ASSIGNABLE_CASES = [
```

Comment:
I'd like to see a few more test cases of the existing behavior:
* `String` is not assignable to `DynamicArray[Byte]` or `DynamicBytes`, and the reverse
* `Address` is not assignable to `StaticArray[Byte, 32]` or `StaticBytes[32]`, and the reverse
* `StaticBytes[x]` is not assignable to `StaticArray[Byte, y]`, where `x != y`

Reply:
These test cases should be added in recent commits.